### ISSUE TITLE
Simplifying directory layout

### ISSUE THREAD

**Issue by longhotsummer** [censusreporter/censusreporter#144]
Hi. We're working on using censusreporter in a couple of African countries and often the first stumbling block with a new team is that the directory layout is different to most Django projects and folks get confused about having to add things to the python path and setting DJANGO_SETTINGS_MODULE.

I propose some changes that would make censusreporter Just Work (tm) for most users familiar with Django and virtualenv, bring it in line with other Django projects I've seen, and help with this confusion:

1. Move the `census` app directory from `censusreporter/apps/census` into the root directory.
2. Move the `bin` and `config` directories into the root directory.

I'd like to eliminate having to set the DJANGO_SETTINGS_MODULE too -- by far the bulk of the time people are interacting with it in dev mode so I think that should be the default. Two possibilities:

1. Unify all settings into `census/settings.py`, `census/urls.py` and `census/wsgi.py` and for the few differences (there really are very few) between dev and prod use an `if` statement [like in this example](https://github.com/Code4SA/indigo/blob/master/indigo/settings.py#L23).
2. Keep the configs separate and change `manage.py` and `wsgi.py` so that they default to the dev configs and handle the prod exception in prod.

My preference is a single settings file. I've found over time that having just one place to find project settings makes it easier to work out what the actual settings are, reduces confusion and avoids the mistake of applying a setting in dev but not prod.

I've tested both the directory change and the first settings option and it's straightforward. No code needed to change for the directory layout adjustment, and the differences between prod and dev settings are minor.

Now, I know we can do this in our fork (and probably will), but I think upstream could benefit from it too. It would also simplify our process when sending pull requests and pulling in updates from upstream, since most files have now moved.

What are your thoughts?

**Comment by JoeGermuska** [comment]
@iandees has been our deployment wrangler, so I'd like his input, but I agree with your rationale

**Comment by ryanpitts** [comment]
Yes, also very interested in looking at the packaging for the app. If we can find simplifications, that seems great.

**Comment by longhotsummer** [comment]
@iandees any thoughts?

**Comment by JoeGermuska** [comment]
So I had some time to take a look, and I definitely agree with the principle of aligning local development with Django standards and making deployed systems the exceptional case.

I have reservations about merging `settings.py` into one file -- stuff like [this configuration for the API_URL and the caching](https://github.com/censusreporter/censusreporter/blob/master/censusreporter/config/dev/settings.py#L13-L25) seems awkward to control with `if/else` -- having the deployed systems "extend" the default config instead of requiring special handling for local dev.

Totally agree with merging `urls.py`

Pretty Indifferent about `wsgi.py` but my instinct is also to keep them separate since there's no access to any kind of flag in settings.py and if there are important differences, they are more likely to be more complicated than I like to see in if/else statements.

@iandees said he should be able to weigh in later today.

**Comment by longhotsummer** [comment]
Moving the directories around to align with common django standards will have the most beneficial impact I feel, so I'm happy with doing that but leave settings.py separated out.

**Comment by iandees** [comment]
From a deploy standpoint the directory structure doesn't matter very much. The most important aspect of this change is development and maintenance, which I think @ryanpitts should have the final say on.

**Comment by JoeGermuska** [comment]
I think @ryanpitts expressed his sympathy with the idea above, and since he's in a :zap:-challenged state right now, no need to wait on his followup.

**Comment by longhotsummer** [comment]
Ok, we're going ahead with moving things into the root directory. I'll send a pull request that matches what we've done on our side.

**Comment by longhotsummer** [comment]
I've created https://github.com/censusreporter/censusreporter/pull/148 which duplicates what we have done for our fork.
